com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -59,141 +65,102 @@ internal struct SnapshotSpawnCommand
         internal Quaternion ObjectRotation;
         internal Vector3 ObjectScale;
 
-        // snapshot internal
+        // internal
         internal int TickWritten;
-        internal List<ulong> TargetClientIds;
-        internal int TimesWritten;
-
-        // for Metrics
-        internal NetworkObject NetworkObject;
     }
 
-    internal class ClientData
+    internal struct SnapshotSpawnCommandMeta
     {
-        internal struct SentSpawn // this struct also stores Despawns, not just Spawns
-        {
-            internal ulong SequenceNumber;
-            internal ulong ObjectId;
-            internal int Tick;
-        }
-
-        internal ushort SequenceNumber = 0; // the next sequence number to use for this client
-        internal ushort LastReceivedSequence = 0; // the last sequence number received by this client
-        internal ushort ReceivedSequenceMask = 0; // bitmask of the messages before the last one that we received.
+        internal List<ulong> TargetClientIds;
+    };
 
-        internal int NextSpawnIndex = 0; // index of the last spawn sent. Used to cycle through spawns (LRU scheme)
-        internal int NextDespawnIndex = 0; // same as above, but for despawns.
+    internal struct ClientData
+    {
+        internal int LastReceivedTick; // the last tick received by this client
 
         // by objectId
         // which spawns and despawns did this connection ack'ed ?
-        internal Dictionary<ulong, int> SpawnAck = new Dictionary<ulong, int>();
+        internal Dictionary<ulong, int> SpawnDespawnAck;
```

Comment:
Would it be potentially valuable to make this `NativeHashMap<ulong, int>`?

Reply:
I think we'd like to figure out our Jobs/Collections strategy we'll follow within the SDK but starting to use Collections stuff earlier would be nice, good-to-have but still optional atm — just FWIW, FYI.